src/dlr_pipeline.cc

```diff
@@ -0,0 +1,182 @@
+#include "dlr_pipeline.h"
+
+#include <stdlib.h>
+#include <cstring>
+#include <fstream>
+#include <iterator>
+#include <numeric>
+
+using namespace dlr;
+
+
+void PipelineModel::CheckModelsCompatibility(const DLRModelPtr& pm, const DLRModelPtr& cm,
+                                             const int cm_id, const bool isRuntimeCheck) {
+  if (!isRuntimeCheck) {
+    CHECK_EQ(pm->GetNumOutputs(), cm->GetNumInputs())
+      << "Number of outputs/inputs mismatch between models"
+      << cm_id - 1 << " and " << cm_id << std::endl;
+  }
+  // Check each model input
+  for (int j = 0; j < cm->GetNumInputs(); j++) {
+    int64_t out_size;
+    int out_dim;
+    pm->GetOutputSizeDim(j, &out_size, &out_dim);
+    if (!isRuntimeCheck) {
+      CHECK_EQ(out_dim, cm->GetInputDim(j))
+        << "Number of dimensions mismatch between output/input #" << j << ", models "
+        << cm_id - 1 << " and " << cm_id << std::endl;
+    }
+    const int64_t in_size = cm->GetInputSize(j);
```

Comment:
We should change GetInputSize() to always return -1 for RelayVM if any one of the input shapes is -1. Currently, it is not guaranteed to be negative since it just multipleis all input shapes.

Reply:
opened new PR for that https://github.com/neo-ai/neo-ai-dlr/pull/259